src/Enemy/EnemyStateWander.cpp

```diff
@@ -0,0 +1,107 @@
+#include "Enemy/EnemyStateWander.h"
+#include <al/Library/LiveActor/ActorActionFunction.h>
+#include <al/Library/LiveActor/ActorCollisionFunction.h>
+#include <al/Library/LiveActor/ActorMovementFunction.h>
+#include <al/Library/LiveActor/ActorPoseKeeper.h>
+#include <al/Library/Math/MathRandomUtil.h>
+#include <al/Library/Nerve/NerveSetupUtil.h>
+#include <al/Library/Nerve/NerveUtil.h>
+
+namespace {
+NERVE_IMPL(EnemyStateWander, Wait);
+NERVE_IMPL(EnemyStateWander, Walk);
+NERVE_IMPL(EnemyStateWander, Fall);
+
+struct {
+    NERVE_MAKE(EnemyStateWander, Wait);
+    NERVE_MAKE(EnemyStateWander, Walk);
+    NERVE_MAKE(EnemyStateWander, Fall);
+} NrvEnemyStateWander;
+
+}  // namespace
+
+EnemyStateWander::EnemyStateWander(al::LiveActor* actor, const char* stateName) : al::ActorStateBase("さんぽ状態", actor), mStateName(stateName) {
+    initNerve(&NrvEnemyStateWander.Wait, 0);
+};
+
+void EnemyStateWander::appear() {
+    mIsDead = false;
+    al::setNerve(this, &NrvEnemyStateWander.Walk);
+}
+
+void EnemyStateWander::exeWait() {
+    if (al::isFirstStep(this)) {
+        al::startAction(mActor, "Wait");
+        mRandNum = al::getRandom(60);
+    }
+    if (!al::isGreaterEqualStep(this, mRandNum + 120)) {
+        if (al::isOnGround(mActor, 0)) {
+            if (al::isFallNextMove(mActor, al::getVelocity(mActor), 50.0f, 200.0f)) {
+                al::scaleVelocity(mActor, -1.0f);
+                al::setVelocityY(mActor, 0.0f);
+            }
+            al::scaleVelocity(mActor, 0.7f);
+            al::addVelocityToGravityFittedGround(mActor, 2.0f, 0);
+        } else
+            al::setNerve(this, &NrvEnemyStateWander.Fall);
+    } else
+        al::setNerve(this, &NrvEnemyStateWander.Walk);
```

Comment:
Change this to early return or the function below to a similar structure like this one